cmds/minigames/connect4.js

```diff
@@ -175,288 +289,320 @@ class connect4 extends two_p_Game{
-            if(this.stack[i] === -1){
-                countstack++
+        // check how many pieces in a column
+        let countstack = 0;
+        for (let i = 0; i < 7; i++) {
+            if (this.stack[i] === -1) {
+                countstack++;
             }
         }
 
-        if(countstack === 7) return 3;
-
-        return 1
+        // a column is full
+        if (countstack === 7) return 3;
 
+        return 1;
     }
 
-    
-
-    search(x, y, color, dx, dy, count, board = this.map){
-
-        if(x<0 || x>6 || y<0 || y>5) return count
-        if(color !== board[y][x]) return count
-
-        return this.search(x+dx, y+dy, color, dx, dy, count+1, board)
+    // (not sure what this whole code does)
+    search(x, y, color, dx, dy, count, board = this.map) {
+        if (x < 0 || x > 6 || y < 0 || y > 5) return count;
+        if (color !== board[y][x]) return count;
 
+        return this.search(x + dx, y + dy, color, dx, dy, count + 1, board);
     }
 
-    search2(x, y, color, dx, dy, count, board){
-
-        if(x<0 || x>6 || y<0 || y>5) return {count : count, live : 0}
-        if(board[y][x] !== color){
-            if(board[y][x] === "⚪"){
-                return {count : count, live : 1}
+    search2(x, y, color, dx, dy, count, board) {
+        if (x < 0 || x > 6 || y < 0 || y > 5) return { count: count, live: 0 };
+        if (board[y][x] !== color) {
+            if (board[y][x] === "⚪") {
+                return { count: count, live: 1 };
             }
-            return {count : count, live : 0}
+            return { count: count, live: 0 };
         }
 
-        let result = this.search2(x+dx, y+dy, color, dx, dy, count+1, board)
-
-        return result
-
+        let result = this.search2(
+            x + dx,
+            y + dy,
+            color,
+            dx,
+            dy,
+            count + 1,
+            board
+        );
+
+        return result;
     }
 
-    c4minimax(board, stack, depth, max, lx, ly){
-
-        if(this.check(lx,ly,board) === 2 || depth === 0){
-
-            let temp1 = this.board_evaluation(board,"🟡")
-            let temp1score = temp1.two + temp1.three*5/3 + temp1.four*15
-            let temp2 = this.board_evaluation(board,"🔴")
-            let temp2score = temp2.two + temp2.three*5/3 + temp2.four*20
-            if(temp1.four>0 || temp2.four>0) console.log(temp1score-temp2score)
-
-            return {score : temp1score-temp2score, move : 0}
+    // calculate the best move
+    c4minimax(board, stack, depth, max, lx, ly) {
+        if (this.check(lx, ly, board) === 2 || depth === 0) {
+            let temp1 = this.board_evaluation(board, "🟡");
+            let temp1score =
+                temp1.two + (temp1.three * 5) / 3 + temp1.four * 15;
+            let temp2 = this.board_evaluation(board, "🔴");
+            let temp2score =
+                temp2.two + (temp2.three * 5) / 3 + temp2.four * 20;
+            if (temp1.four > 0 || temp2.four > 0)
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/xiaomiao-fong/Chen_bot/pullRequest?prid=7807535)
```suggestion
            if (temp1.four > 0 || temp2.four > 0)
```